installer_fabfile.py

```diff
@@ -293,16 +298,21 @@ def cloud_init(gs_tenant,
             if stack_password is None:
                 stack_password = prompt(
                     cyan("Enter Openstack user password: "))
+
             if stack_auth_url is None:
-                while stack_auth_url is None or \
-                        not stack_auth_url.endswith(CLOUD_AUTH_URL_VERSION):
-                    stack_auth_url = prompt(
-                        cyan("Enter OpenStack auth URL base "
-                             "(ex: http://10.10.10.10:5000/v2.0): "))
-                    if not stack_auth_url.endswith(CLOUD_AUTH_URL_VERSION):
-                        stack_auth_url = re.sub('/v2.0[/]?$|/v3[/]?$',
-                                                CLOUD_AUTH_URL_VERSION,
-                                                stack_auth_url)
+                stack_auth_url = \
+                    prompt(cyan("Enter OpenStack auth URL "
+                                "(eg: http://10.10.10.10:5000/): "))
+
+                if re.search(AUTH_URL_VERSION_LIKELY, stack_auth_url[-9:]):
+                    # The user shouldn't have included the version segment, but
+                    # did anyway. Remove it.
+                    version_index = re.search(AUTH_URL_VERSION_LIKELY,
+                                              stack_auth_url)
+                    stack_auth_url = stack_auth_url[:version_index.start()]
+
+                # Append our version number to the base URL.
+                stack_auth_url = os.path.join(stack_auth_url, AUTH_URL_VERSION)
```

Comment:
The code in master breaks when the user doesn't enter a version number. Or, enters any version number other than exactly "2.0" or "3":

![screen shot 2015-06-04 at 7 41 25 am](https://cloud.githubusercontent.com/assets/231148/7986584/35125724-0a8d-11e5-9aeb-4ce2421ce7c9.png)

![screen shot 2015-06-04 at 7 43 18 am](https://cloud.githubusercontent.com/assets/231148/7986636/7f10ae98-0a8d-11e5-93ce-37e602ec39ae.png)


The code in this pull request will work regardless of whether the entered URL has a version number segment, and regardless of what that version number is.